openshift/dockerfiles/nginx.conf

```diff
@@ -0,0 +1,98 @@
+# For more information on configuration, see:
+#   * Official English Documentation: http://nginx.org/en/docs/
+#   * Official Russian Documentation: http://nginx.org/ru/docs/
+
+
+worker_processes auto;
+error_log /var/log/nginx/error.log;
+pid /run/nginx.pid;
+
+# Load dynamic modules. See /usr/share/doc/nginx/README.dynamic.
+include /usr/share/nginx/modules/*.conf;
+
+events {
+    worker_connections 1024;
+}
+
+http {
+    # perl_modules /opt/app-root/etc/perl;
+    # perl_require Version.pm;
+    # perl_set $perl_version Version::installed;
+
+    log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
+                      '$status $body_bytes_sent "$http_referer" '
+                      '"$http_user_agent" "$http_x_forwarded_for"';
+
+
+    sendfile            on;
+    tcp_nopush          on;
+    tcp_nodelay         on;
+    keepalive_timeout   65;
+    types_hash_max_size 2048;
+
+    include             /etc/nginx/mime.types;
+    default_type        application/octet-stream;
+
+    # Load modular configuration files from the /etc/nginx/conf.d directory.
+    # See http://nginx.org/en/docs/ngx_core_module.html#include
+    # for more information.
+    include /opt/app-root/etc/nginx.d/*.conf;
+
+    server {
+        listen       8080 default_server;
+        listen       [::]:8080 default_server;
+        server_name  _;
+        root         /opt/app-root/src;
+		add_header      Strict-Transport-Security "max-age=31536000;";
+		add_header      X-XSS-Protection "1; mode=block";
+		add_header      X-Content-Type-Options "nosniff";
+		add_header      X-Frame-Options "DENY";
+		gzip            on;
+		server_tokens   off;
+
+        # Load configuration files for the default server block.
+        include /opt/app-root/etc/nginx.default.d/*.conf;
+
+        location /api/v3/nr_measurements {
+			gzip                off;
+			proxy_http_version  1.1;
+			proxy_read_timeout  24h;
+			proxy_set_header    Connection "";
+			proxy_pass          http://externalserver:5001/api/v3/nr_measurements;
```

Comment:
In the docker compose we use `external_server` as server name. Is there a specific reason why it's `external server` here? If not, I'd prefer to keep the service names the same.

Reply:
The underscore is not allowed for service connection in OpenShift. Not sure if it is company specific or a Red Hat standard